compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -8,32 +8,50 @@
 #include "iree-amd-aie/IR/AMDAIEOps.h"
 #include "iree-amd-aie/Transforms/Passes.h"
 #include "iree-amd-aie/Transforms/Transforms.h"
+#include "iree/compiler/Codegen/TransformStrategies/GPU/Common.h"
 #include "mlir/Dialect/SCF/Transforms/Transforms.h"
 #include "mlir/IR/Iterators.h"
 #include "mlir/Pass/Pass.h"
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 #include "mlir/Transforms/LoopInvariantCodeMotionUtils.h"
 
-#define DEBUG_TYPE "iree-amdaie-unroll-and-distribute-workgroup"
+#define DEBUG_TYPE "iree-amdaie-distribute-cores-and-objectfifos"
 
 namespace mlir::iree_compiler::AMDAIE {
 
+// Used to annotate loops that should be unrolled.
+static const llvm::StringLiteral kAMDAIELoopUnroll = "amdaie.unroll";
+
 namespace {
 
-/// Convert scf.forall ops within a workgroup to scf.for ops
-/// TODO(jornt): use upstream `forallToFor` function once merged.
-LogicalResult workgroupForallToFor(ModuleOp moduleOp) {
+/// Convert local scf.forall ops chosen for parallel distribution to scf.for
+/// ops.
+LogicalResult localForallToFor(ModuleOp moduleOp) {
   IRRewriter rewriter(moduleOp.getContext());
-  WalkResult res = moduleOp->walk([&](AMDAIE::WorkgroupOp workgroupOp) {
-    WalkResult workgroupRes = workgroupOp->walk([&](scf::ForallOp forallOp) {
-      if (failed(scf::forallToForLoop(rewriter, forallOp))) {
-        workgroupOp.emitOpError()
-            << "failed to transform scf.forall to scf.for";
+  WalkResult res = moduleOp->walk([&](scf::ForallOp forallOp) {
+    SmallVector<Attribute> mapping =
+        llvm::to_vector(forallOp.getMapping()->getValue());
+    // We index on thread mapping for core and dma unrolling and buffer
+    // distribution.
+    if (!isa<mlir::gpu::GPUThreadMappingAttr>(*mapping.begin()))
+      return WalkResult::advance();
+
+    SmallVector<Operation *> results;
+    if (failed(scf::forallToForLoop(rewriter, forallOp, &results))) {
+      forallOp.emitOpError() << "failed to transform scf.forall to scf.for";
+      return WalkResult::interrupt();
+    }
+    // Set attribute to unroll this loop later in this pass.
+    for (Operation *res : results) {
```

Comment:
Nit: I would avoid using the same variable name "res" here.

Reply:
Great catch! Thanks